### PR TITLE
Add OAuth contoller and GitHubService.

### PR DESCRIPTION
**Pull request by dcharoulis**
Improved comments in project.

### CODE REVIEW

File: app/controllers/OAuthGitHubController.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }).getOrElse(Future.successful(BadRequest("No parameters supplied")))
+  }
+
+  /**
+    * If authentication was successful returns the projects in GitHub.
+    * @return
+    */
+  def success() = Action.async { request =>
+    request.session.get("oauth-token").fold(Future.successful(Unauthorized("No way Jose"))) { authToken =>
+      cacheApi.set("authToken", authToken)
```
Comment (erikwj) [inline app/controllers/OAuthGitHubController.scala]: When does the cache expire?

File: app/services/GitHubService.scala [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                   collaborators_url:String,
+                   teams_url:String,
+                   hooks_url:String,
+                   issue_events_url:String,
+                   events_url:String,
+                   assignees_url:String,
+                   branches_url:String,
+                   tags_url:String,
+                   blobs_url:String,
+                   //git_tags_url:String,
```
Comment (erikwj) [inline app/services/GitHubService.scala]: can these commented-out fields be deleted?

Comment (dcharoulis) [reply app/services/GitHubService.scala]: Sure, i am going to remove the commented ones. Actually maybe we can remove more in the future if they are not going to be used.

File: app/controllers/OAuthGitHubController.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    * If authentication was successful returns the projects in GitHub.
+    * @return
+    */
+  def success() = Action.async { request =>
+    request.session.get("oauth-token").fold(Future.successful(Unauthorized("401 Authentication token not found!"))) { authToken =>
+      Future(Redirect(TEAHubController.githubRepositories())
+      )
+    }
+  }
+}
```
Comment (mahyaa) [inline app/controllers/OAuthGitHubController.scala:63]: Why there is no documentation for what does the method returns? (for all of the methods in this class)

**Pull request merged**